### PR TITLE
PWN-6441 - block ui for off-ramp

### PR DESCRIPTION
**Pull request by kabramovp2p**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-6441

## Description of Work

PWN-6441 - block ui for off-ramp

### CODE REVIEW

**Review by NikolaiEmelianov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/moonpay/ui/transaction/SellTransactionDetailsBottomSheet.kt [hunk 1]
```diff
@@ -214,6 +215,21 @@ class SellTransactionDetailsBottomSheet :
         binding.layoutDetails.buttonRemoveOrCancel.isLoadingState = isLoading
     }
 
+    override fun navigateToSendScreen(
+        tokenToSend: Token.Active,
+        sendAmount: BigDecimal,
+        receiverAddress: String
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/moonpay/ui/transaction/SellTransactionDetailsBottomSheet.kt:221]: Maybe you will pass `SearchResult.AddressFound` here?

File: app/src/main/java/org/p2p/wallet/sell/ui/lock/SellLockedFragment.kt [hunk 2]
```diff
@@ -133,6 +132,21 @@ class SellLockedFragment :
         popBackStackTo(MainFragment::class)
     }
 
+    override fun navigateToSendScreen(
+        tokenToSend: Token.Active,
+        sendAmount: BigDecimal,
+        receiverAddress: String
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/sell/ui/lock/SellLockedFragment.kt:138]: same

**Review by 1sakov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/newsend/ui/NewSendPresenter.kt [hunk 3]
```diff
@@ -153,6 +155,14 @@ class NewSendPresenter(
...
 
             initializeFeeRelayer(view, initialToken, solToken)
+            initialAmount?.let { inputAmount ->
+                if (calculationMode.getCurrencyMode() is CurrencyMode.Fiat.Usd) {
+                    switchCurrencyMode()
+                }
+                view.updateInputValue(inputAmount.scaleShort().toPlainString(), forced = true)
+                calculationMode.updateInputAmount(inputAmount.scaleShort().toPlainString())
+                view.disableInputs()
+            }
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/newsend/ui/NewSendPresenter.kt:165]: could you please move to a separate function? 
Something like `setupDefaultFields`

File: app/src/main/java/org/p2p/wallet/moonpay/ui/transaction/SellTransactionDetailsBottomSheet.kt (hunk 1 above)
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/moonpay/ui/transaction/SellTransactionDetailsBottomSheet.kt:221]: but it does

Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/moonpay/ui/transaction/SellTransactionDetailsBottomSheet.kt:221]: val recipient = SearchResult.AddressFound(AddressState(receiverAddress)) 
just a line down from this code

**Pull request merged**